### PR TITLE
PWGHF: Add downscaling to Lc tree creator

### PR DESCRIPTION
**Pull request by AnnalenaKa**
Add a configurable to set the downscaling factor for the Lc tree creator. 
Only this fraction of candidates will be stored in the tree.

### CODE REVIEW

**Comment by fcolamar** [comment]
Hi @AnnalenaKa, is there a reason for removing the check on the mass hypothesis and filling only the case corresponding to Lc->piKp, evaluating the invariant mass always under that assumption?
Also (minor) you may want to fix the typo on donwSampleBkgFactor, apart from that your PR is fine to me.

File: PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx [hunk 1]
```diff
@@ -211,12 +214,13 @@ struct CandidateTreeWriter {
...
       auto trackPos2 = candidate.index2_as<aod::BigTracksPID>(); // positive daughter (negative for the antiparticles)
       auto fillTable = [&](int CandFlag,
-                           int FunctionSelection,
+                           // int FunctionSelection,
                            float FunctionInvMass,
                            float FunctionCt,
                            float FunctionY,
                            float FunctionE) {
-        if (FunctionSelection >= 1) {
+        double pseudoRndm = trackPos1.pt() * 1000. - (long)(trackPos1.pt() * 1000);
```
Comment (vkucera) [inline PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx:222]: Is this a good way of getting a pseudorandom number?

File: PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx [hunk 2]
```diff
@@ -184,6 +184,9 @@ struct CandidateTreeWriter {
   {
   }
 
+  Configurable<int> isSignal{"isSignal", 1, "save only MC matched candidates"};
+  Configurable<double> donwSampleBkgFactor{"donwSampleBkgFactor", 1., "Fraction of candidates to store in the tree"};
```
[the remark this answers is not in the document]
Comment (AnnalenaKa) [reply PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx]: Yes,  that was a typo. I corrected it.

**Comment by fcolamar** [comment]
For me the PR is ok, unless @vkucera has further comments I would approve.
Fabio

**Review by fcolamar — APPROVED** [review]

File: PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx [hunk 3]
```diff
@@ -216,7 +219,8 @@ struct CandidateTreeWriter {
                            float FunctionCt,
                            float FunctionY,
                            float FunctionE) {
-        if (FunctionSelection >= 1) {
+        double pseudoRndm = trackPos1.pt() * 1000. - (long)(trackPos1.pt() * 1000);
+        if (FunctionSelection >= 1 && std::abs(candidate.flagMCMatchRec()) >= isSignal && pseudoRndm < downSampleBkgFactor) {
```
Comment (vkucera) [inline PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx:223]: This is wrong.
```c++
        std::abs(candidate.flagMCMatchRec()) >= isSignal
```
`flagMCMatchRec` returns a bitmap which will be >= 1 for all 3-prong decay channels.
You have to test the right bit if you want to select only Lc.

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx (hunk 3 above)
Comment (AnnalenaKa) [reply PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx]: Dear Vit, 
thanks for pointing this out. Indeed this check did not select the Lc signal. I added the test on the LctoPKpi in the selection now.

**Review by vkucera — APPROVED** [review]

**Pull request merged**